src/core/ddsc/CMakeLists.txt

```diff
@@ -49,7 +48,6 @@ PREPEND(hdrs_public_ddsc "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include/dd
     ddsc/dds_public_qos.h
     ddsc/dds_public_qosdefs.h
     ddsc/dds_public_status.h
-    ddsc/dds_rhc.h
```

Comment:
Definitely, I tried to implement the custom RHC in rmw_cyclonedds_cpp and got lots of compile error. I think move them to *dds/ddsc/dds_public_*.h" and make their name with *dds/ddsc/dds_public_rhc.h* and "dds/ddsc/dds_public_whc.h" might be a good idea.

Reply:
Something like that would make sense, but there are still some rough edges in the interface (like dealing with read/query conditions). There is also the complication that some of the interface comes from the ddsi layer and some of it from the ddsc one, and it isn't obvious how that would be dealt with best in the header files.

I'd rather stick to the current situation until those complications have been dealt with, but I'm open to arguments why it would be better to not wait.